sympy/core/relational.py

```diff
@@ -542,10 +542,56 @@ def _eval_simplify(self, **kwargs):
                     enew = e.func(m*x, -b)
                 measure = kwargs['measure']
                 if measure(enew) <= kwargs['ratio']*measure(e):
-                    e = enew
+                    e = enew.canonical
             except ValueError:
                 pass
-        return e.canonical
+
+        # Simplify expressions of the type Eq(x, Min/Max(..., x, ...))
+        from sympy.functions.elementary.miscellaneous import Min, Max
+        if self.has(Min, Max):
+            minmax, constant = None, None
+            if isinstance(e.rhs, (Min, Max)) and not e.lhs.has(Min, Max):
+                minmax, constant = e.rhs, e.lhs
+            elif isinstance(e.lhs, (Min, Max)) and not e.rhs.has(Min, Max):
+                minmax, constant = e.lhs, e.rhs
+            if minmax:
+                from sympy.utilities.iterables import sift
+                enew = None
+                if constant.is_number:
+                    numberpart, remaining = sift(minmax.args, lambda x: x.is_number, binary=True)
+                    if numberpart:
+                        f = minmax.func
+                        if minmax.func == Min:
```

Comment:
```suggestion
                        if isinstance(minmax, Min):
```

Reply:
Yes, there are several inconsistencies. Had some problems getting things to work, so the code is a bit rough at places. Will do an overhaul today as it is working now.